src/presentation/test/integration/registration/generalPartner/address/confirm-general-partner-usual-residential-address.test.ts

```diff
@@ -0,0 +1,123 @@
+import request from "supertest";
+import enTranslationText from "../../../../../../../locales/en/translations.json";
+import cyTranslationText from "../../../../../../../locales/cy/translations.json";
+import { getUrl, setLocalesEnabled, testTranslations } from "../../../../utils";
+import { CONFIRM_GENERAL_PARTNER_USUAL_RESIDENTIAL_ADDRESS_URL } from "../../../../../controller/addressLookUp/url";
+import { LIMITED_PARTNERS_URL } from "../../../../../controller/registration/url";
+import app from "../../../app";
+import { appDevDependencies } from "../../../../../../config/dev-dependencies";
+import AddressPageType from "../../../../../controller/addressLookUp/PageType";
+import { ApiErrors } from "../../../../../../domain/entities/UIErrors";
+import GeneralPartnerBuilder from "../../../../builder/GeneralPartnerBuilder";
+
+describe("Confirm General Partner Usual Residential Address Page", () => {
+  const URL = getUrl(CONFIRM_GENERAL_PARTNER_USUAL_RESIDENTIAL_ADDRESS_URL);
+
+  beforeEach(() => {
+    setLocalesEnabled(false);
+    appDevDependencies.cacheRepository.feedCache({
+      [appDevDependencies.transactionGateway.transactionId]: {
+        ["usual_residential_address"]: {
+          postal_code: "ST6 3LJ",
+          premises: "4",
+          address_line_1: "line 1",
+          address_line_2: "line 2",
+          locality: "stoke-on-trent",
+          region: "region",
+          country: "GB-ENG"
+        }
+      }
+    });
+
+    const generalPartner = new GeneralPartnerBuilder()
+      .withId(appDevDependencies.generalPartnerGateway.generalPartnerId)
+      .isPerson()
+      .build();
+
+    appDevDependencies.generalPartnerGateway.feedGeneralPartners([generalPartner]);
+  });
+
+  describe("GET Confirm Usual Residential Address Page", () => {
+    it("should load the confirm usual residential address page with English text", async () => {
+      setLocalesEnabled(true);
+
+      const res = await request(app).get(URL + "?lang=en");
+
+      expect(res.status).toBe(200);
+      testTranslations(res.text, enTranslationText.address.confirm.usualResidentialAddress);
+      expect(res.text).not.toContain("WELSH -");
+
+      expect(res.text).toContain("4 Line 1");
```

Comment:
Should these assertions also be present in the Welsh language test, immediately below?

Reply:
True. Template code and mechanics of getting the values displayed on the page are the same, so probably fine.